docs/source/spec-workload-executor.rst

```diff
@@ -58,179 +58,227 @@ After accepting the inputs, the workload executor:
    to communicate to the workload executor, and ultimately the unified test
    runner, that they should stop running operations.
 
+#. MUST initialize the following variables, which will later be used to generate
+   the ``results.json`` and ``events.json`` output files:
+
+   * ``events``: Empty array of objects.
+
+   * ``errors``: Empty array of objects.
+
+   * ``failures``: Empty array of objects.
+
+   * ``numIterations``: Integer with value -1.
+
+   * ``numSuccesses``: Integer with value -1.
+
+   Note: ``numErrors`` and ``numFailures`` are intentionally omitted here as
+   they will be derived directly from ``errors`` and ``failures``.
```

Comment:
Initializing all of the output fields for the JSON files up front avoids the ambiguity about what happens when the test runner fails. Now, we always have _some_ value to use for all of these fields, and the error/failure counts are always calculated from the corresponding arrays.